### ISSUE TITLE
Jest addon does not work

### ISSUE THREAD

**Issue by tmeasday** [storybooks/storybook#5197]
See examples in official storybook.

**Comment by alflennik** [comment]
Confirmed on `@storybook/react` and `@storybook/addon-jest` versions `5.0.0-beta.1`. 

I am glad this is high priority, but I see there has been no progress in three weeks.

Since I need both Jest support and Emotion v10, which is only supported in 5.0.0, my usecase for Storybook is broken until this is fixed.

**Comment by tmeasday** [comment]
Hi @alflennik!

A couple of questions:

1. When you say you need Jest support--do you mean the `addon-jest`? You can certainly use jest and storyshots without it (just FYI)

2. I am not sure that you cannot use Emotion 10 with the latest Storybook 4.1 release. I think you should be able to. What issue are you seeing?

3. We are happy to take contributions here! We haven't looked at this addon yet because it is not very commonly used, but I suspect the issue may be similar to the one fixed in this PR:  https://github.com/storybooks/storybook/pull/5452

**Comment by shilman** [comment]
Confirmed and will investigate/fix

**Comment by shilman** [comment]
Zoinks!! I just released https://github.com/storybooks/storybook/releases/tag/v5.0.0-rc.2 containing PR #5644 that references this issue. Upgrade today to try it out!

Because it's a pre-release you can find it on the `@next` NPM tag.

Closing this issue. Please re-open if you think there's still more to do.

**Comment by alflennik** [comment]
First, thank you so much for prioritizing this, but I am still having issues!

The TL;DR is:  **Has this been tested with the filesExt parameter?**

I just upgraded to storybook and addon-jest version 5.0.0-rc.7 and noticed that the test panel in Storybook remained blank. So I removed all the addon-jest code in my repo and started reinstalling it following the instructions [in the docs](https://github.com/storybooks/storybook/tree/next/addons/jest).

I got to the hello world section, where you manually link the test file, and yes, the tests appeared!

```js
// Boilerplate addon-jest
import results from '../.jest-test-results.json';
import { withTests } from '@storybook/addon-jest';

storiesOf('MyComponent', module)
  .addDecorator(withTests({ results }))
  .add(
    'This story shows test results from MyComponent.test.js and MyOtherComponent.test.js',
    () => <div>Jest results in storybook</div>,
    {
      jest: ['MyComponent.test.js', 'MyOtherComponent.test.js'],
    }
  );
```

Then I set up the `addDecorator` feature in config.js like this:

```js
addDecorator(
  withTests({
    results,
  })
);
```

The tests disappeared. But since I'm using a jsx file extension on my tests, I updated it to:

```js
addDecorator(
  withTests({
    results,
    filesExt: ".test.jsx"
  })
);
```

The tests did not reappear as expected.

**Comment by alflennik** [comment]
I got halfway through a minimum repro repo and I discovered that some, ahem, abuses of the Storybook API that I was doing in the config.js file were the cause of my issue.

My workaround was functional in Storybook 4.0 but broken in 5.0.

So it looks like the fix to addon-jest is good to go! And thanks so much for the responsiveness!

**Comment by tmeasday** [comment]
Awesome! Anything we need to add to the migration guide @alflennik?

**Comment by alflennik** [comment]
Nope!
